### PR TITLE
[MM-51487] allow uppercase team/channel names for import

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
We are quite forgiving on case sensitivity on other entities such as `User` and I think we can continue to be like that for team and channel names.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-51487

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/import_functions.go [hunk 1]
```diff
@@ -189,15 +189,16 @@ func (a *App) importTeam(rctx request.CTX, data *imports.TeamImportData, dryRun
...
 	var team *model.Team
-	team, err := a.Srv().Store().Team().GetByName(*data.Name)
+	team, err := a.Srv().Store().Team().GetByName(teamName)
 
 	if err != nil {
 		team = &model.Team{}
 	}
 
-	team.Name = *data.Name
+	team.Name = teamName
```
Comment (agnivade) [inline server/channels/app/import_functions.go:201]: Do we really need to modify this again? We are getting the team by `teamName`, so it is given that the team name will be `teamName`.

File: server/channels/app/import_functions.go [hunk 2]
```diff
@@ -264,22 +265,25 @@ func (a *App) importChannel(rctx request.CTX, data *imports.ChannelImportData, d
...
-	if result, gErr := a.Srv().Store().Channel().GetByNameIncludeDeleted(team.Id, *data.Name, true); gErr == nil {
+	if result, gErr := a.Srv().Store().Channel().GetByNameIncludeDeleted(team.Id, channelName, true); gErr == nil {
 		channel = result
 	} else {
 		channel = &model.Channel{}
 	}
 
 	channel.TeamId = team.Id
-	channel.Name = *data.Name
+	channel.Name = channelName
```
Comment (agnivade) [inline server/channels/app/import_functions.go:286]: Same question as above.

**Review by agnivade — APPROVED** [review]
Small comments. Non blocking.

So pleased to finally see these small issues being taken care of. Great work :tada:

File: server/channels/app/import_functions.go (hunk 1 above)
Comment (hanzei) [reply server/channels/app/import_functions.go:201]: If the team can't be found, `team` is an empty struct. Hence, we need to set the Name.

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/channels/app/import_functions.go (hunk 1 above)
Comment (agnivade) [reply server/channels/app/import_functions.go:201]: Ah yes, in that case, maybe we can move this inside the `if` condition?

**Pull request merged**